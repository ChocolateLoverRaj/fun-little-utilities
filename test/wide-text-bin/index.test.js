const iter = require('iter-tools')
const xjest = require('extra-jest')
const { spawnSync } = require('@tools/preloaded-node')
const createWideText = require('wide-text')
const bin = require.resolve('wide-text-bin')

it('--help', xjest.snapSpawn.snap(spawnSync, [bin, '--help']))

describe('when call', () => {
  const inputs = [
    'abc',
    'a b c',
    'hello world',
    'foo bar baz',
    'abc de f'
  ]

  const zipObject = outputs => Array
    .from(iter.zipAll(inputs, outputs))
    .reduce((prev, [key, value]) => Object.assign(prev, { [key]: value }), {})

  const getExpected = options => zipObject(inputs.map(
    text => createWideText(text, options)
  ))

  const getReceived = (argv = []) => zipObject(inputs.map(
    text => String(spawnSync([bin, text, ...argv], { encoding: 'utf8' }).stdout).trim()
  ))

  it('without options', () => {
    const expected = getExpected()
    const received = getReceived()
    expect(received).toEqual(expected)
  })
})
