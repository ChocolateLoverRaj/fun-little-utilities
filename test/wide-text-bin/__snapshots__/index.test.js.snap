// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--help 1`] = `
"
error: null
signal: null
status: 0
stderr: ((EMPTY))
stdout: |+

  index <text> [options]

  Options:
    --version      Show version number                                   [boolean]
    --charSep, -c  Size of separator between characters in a word
                                                             [number] [default: 1]
    --wordSep, -w  Size of separator between words           [number] [default: 2]
    --help         Show help                                             [boolean]


"
`;
